mlfinlab/labeling/excess_over_mean.py

```diff
@@ -0,0 +1,41 @@
+"""
+Return in excess of mean method
+
+Chapter 5, Machine Learning for Factor Investing, by Coqueret and Guida, (2020).
+"""
+import numpy as np
+
+
+def excess_over_mean(prices, binary=False):
+    """
+    Return in excess of mean labeling method. Sourced from Chapter 5.5.1 of Machine Learning for Factor Investing,
+    by Coqueret, G. and Guida, T. (2020).
+
+    Returns a DataFrame containing returns of stocks over the mean of all stocks in the portfolio. Returns a DataFrame
+    of signs of the returns if binary is True. In this case, an observation may be labelled as 0 if it itself is the
+    mean. 
+
+    :param prices: (pd.DataFrame) Close prices of all tickers in the market that are used to establish the mean. NaN
+                    values are ok. Returns on each ticker are then compared to the mean for the given timestamp.
+    :param binary: (bool) If False, the numerical value of excess returns over mean will be given. If True, then only
+                    the sign of the excess return over mean will be given (-1 or 1). A label of 0 will be given if
+                    the observation itself equal to the mean. Note: if there are any 0 labels (as a result of an
+                    observation being exactly equal to the mean for a given time index), np.sign will return a warning,
+                    but the function runs fine.
+    :return: (pd.DataFrame) Numerical returns in excess of the market mean return, or sign of return depending on
+                whether binary is False or True respectively. The last row will be NaN.
+    """
+    # Get return per period
+    returns = prices.pct_change(periods=1).shift(-1)
+
+    # Calculate mean returns for each period as market return
+    market_return = returns.mean(axis=1)
+
+    # Calculate excess over market (mean) return
+    returns_over_mean = returns.sub(market_return, axis=0)
+
+    # If binary is true, returns sign of the return over meam instead of the value
+    if binary:
+        return np.sign(returns_over_mean)
+
+    return returns_over_mean
```

Comment:
I'm proposing a change to this part so that the Warning message wouldn't appear and bother users. (Solution without silencing warnings).

Same can be done for [PR#342](https://github.com/hudson-and-thames/mlfinlab/pull/342) and [PR#351](https://github.com/hudson-and-thames/mlfinlab/pull/351)

</pre>
    if binary:
        to_sign = returns_over_mean.notna().any(axis=1)
        returns_over_mean[to_sign] = np.sign(returns_over_mean[to_sign])

    return returns_over_mean
</pre>

Reply:
I'm still getting the warning after making this change. I did apply the style changes and labell/label thing. Apparently I didn't push the right commit yesterday but it's there now. If I can't find a way to make this warning go away, should I silence it? 

https://github.com/numpy/numpy/issues/15127

Reply:
It doesn't seen that returns_over_mean[to_sign] is any different in regards to NaN values compared to returns_over_mean so np.sign still sees NaNs

Reply:
There is also a case when getting rid of all NaNs would cause issues, id  the data for returns is incomplete and there are NaNs in the data. In that case for those rows we would want to return NaN. Since I know that the final n rows will always be NaN due to the lookforward or lookback period, would it make sense to just manually remove those n rows before returning labels?

Reply:
@ruifanp Thank you for noticing! I was thinking of applying it only on rows where all elements are NaNs, but then if the input data has NaNs the warning will still appear. Found a better solution to work on any input data:

<pre>
if binary:
    returns_over_mean= returns_over_mean.apply(np.sign)

return returns_over_mean
</pre>